src/VisualStudio/Core/Def/Options/VisualStudioOptionStorage.cs

```diff
@@ -115,18 +115,18 @@ public bool TryFetch(LocalUserRegistryOptionPersister persister, OptionKey2 opti
 
     public static readonly IReadOnlyDictionary<string, VisualStudioOptionStorage> Storages = new Dictionary<string, VisualStudioOptionStorage>()
     {
-        {"BlockStructureOptions_CollapseEmptyMetadataImplementationsWhenFirstOpened", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CollapseEmptyMetadataImplementationsWhenFirstOpened")},
-        {"BlockStructureOptions_CollapseImportsWhenFirstOpened", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CollapseImportsWhenFirstOpened")},
-        {"BlockStructureOptions_CollapseMetadataImplementationsWhenFirstOpened", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CollapseMetadataImplementationsWhenFirstOpened")},
-        {"BlockStructureOptions_CollapseRegionsWhenCollapsingToDefinitions", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CollapseRegionsWhenCollapsingToDefinitions")},
-        {"BlockStructureOptions_CollapseRegionsWhenFirstOpened", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CollapseRegionsWhenFirstOpened")},
-        {"BlockStructureOptions_MaximumBannerLength", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.MaximumBannerLength")},
-        {"BlockStructureOptions_ShowBlockStructureGuidesForCodeLevelConstructs", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.ShowBlockStructureGuidesForCodeLevelConstructs")},
-        {"BlockStructureOptions_ShowBlockStructureGuidesForCommentsAndPreprocessorRegions", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.ShowBlockStructureGuidesForCommentsAndPreprocessorRegions")},
-        {"BlockStructureOptions_ShowBlockStructureGuidesForDeclarationLevelConstructs", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.ShowBlockStructureGuidesForDeclarationLevelConstructs")},
-        {"BlockStructureOptions_ShowOutliningForCodeLevelConstructs", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.ShowOutliningForCodeLevelConstructs")},
-        {"BlockStructureOptions_ShowOutliningForCommentsAndPreprocessorRegions", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.ShowOutliningForCommentsAndPreprocessorRegions")},
-        {"BlockStructureOptions_ShowOutliningForDeclarationLevelConstructs", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.ShowOutliningForDeclarationLevelConstructs")},
+        {"dotnet_block_structure_options_collapse_empty_metadata_implementations_when_first_opened", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CollapseEmptyMetadataImplementationsWhenFirstOpened")},
+        {"dotnet_block_structure_options_collapse_imports_when_first_opened", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CollapseImportsWhenFirstOpened")},
+        {"dotnet_block_structure_options_collapse_metadata_implementations_when_first_opened", new RoamingProfileStorage("TextEditor.%LANGUAGE%.Specific.CollapseMetadataImplementationsWhenFirstOpened")},
```

Comment:
❓ If this gets renamed but the other one does not, which unit test will cause the CI build to fail?

Reply:
Are you asking the name in the RoamingProfileStorage? Or are you asking the name where the option is declared?
Like this [one](https://sourceroslyn.io/#Microsoft.CodeAnalysis.LanguageServer.Protocol/Features/Options/BlockStructureOptionsStorage.cs,84755c6207904364,references)?
My understanding it the name here is just an internal ID in our repo, and the persister will use the name to find the storage location in VS, so the name in Option must match the string in this big dictioanry.

We can certainly wait the opinion from @tmat to double check

Reply:
An yeah, there is a test guarding that
https://sourceroslyn.io/#Roslyn.VisualStudio.Next.UnitTests/Options/VisualStudioOptionStorageTests.cs,df788c18beabea68

Reply:
I don't think I could sign off on this unless each string lives in just one place, and the other locations reference it. Also, the dictionary lookup here is unnecessarily expensive. As long as we reference the original string, we could change it to ReferenceEqualityComparer and it wouldn't need to calculate string hashes for each lookup.

Reply:
I see, will modify it later

Reply:
@sharwell  
This PR does not change the representation of the map, just the names of the options.

> I don't think I could sign off on this unless each string lives in just one place, and the other locations reference it.

The test linked above validates that all options in Roslyn have a storage location listed here. It would add unnecessary complexity and no additional value beyond what the test already validates to share the option name between the definition of the option and this map.

> The dictionary lookup here is unnecessarily expensive

I don't see a perf issue here. Each option listed here is deserialized from VS settings at most once and cached cached in IGlobalOptionsService dictionary keyed by OptionKey value thereafter.

Reply:
I broke up and parameterized the tests in #66935